### PR TITLE
refactor rocALUTION documentation

### PR DESCRIPTION
**Pull request by SwRaw**
Refactoring all libraries to adhere to the documentation standards that includes following home page format, index style, language improvements, link fixes, navigation, and overall content readability improvements.

### CODE REVIEW

File: docs/usermanual/linux-installation.rst [hunk 1]
```diff
@@ -1,13 +1,20 @@
...
 -------------
 Prerequisites
 -------------
 
-- A ROCm enabled platform. `ROCm Documentation <https://rocm.docs.amd.com/>`_ has more information on
-  supported GPUs, Linux distributions, and Windows SKUs. It also has information on how to install ROCm.
+A ROCm enabled platform. For information on supported GPUs, Linux distributions, ROCm installation, and Windows SKUs, refer to `ROCm Documentation <https://rocm.docs.amd.com/>`_.
 
 -----------------------------
 Installing pre-built packages
```
Comment (SwRaw) [inline docs/usermanual/linux-installation.rst:20]: "rocALUTION can be installed from `AMD ROCm repository <https://rocm.docs.amd.com/projects/install-on-linux/en/latest/tutorial/quick-start.html>`_. " The link doesn't look right.

File: docs/usermanual/windows-installation.rst [hunk 2]
```diff
@@ -37,11 +39,11 @@ The rocalution.hpp header file must be included in the user code to make calls
...
-Once installed, find rocalution.hpp in the HIP SDK `\\include\\rocalution`
+Once installed, find rocalution.hpp in the HIP SDK ``\\include\\rocalution``
 directory. Only use these two installed files when needed in user code.
 
 ----------------------------------
-Building and Installing rocALUTION
+Building and installing rocALUTION
 ----------------------------------
 
 Building from source is not necessary, as rocALUTION can be used after installing the pre-built packages as described above.
```
Comment (SwRaw) [inline docs/usermanual/windows-installation.rst:56]: "The rocALUTION source code, which is the same as for the ROCm linux distributions, is available at the `rocALUTION github page <https://github.com/ROCmSoftwarePlatform/rocSPARSE>`_." Is this link correct?

File: docs/usermanual/multinode.rst [hunk 3]
```diff
@@ -147,7 +152,7 @@ When the parallel manager, global matrix or global vector are writing to a file,
   rhs.dat.rank.2
   rhs.dat.rank.3
 
-Parallel Manager
+Parallel manager
```
Comment (SwRaw) [inline docs/usermanual/multinode.rst:156]: .. doxygenfunction:: rocalution::ParallelManager::GetGlobalSize
and
.. doxygenfunction:: rocalution::ParallelManager::GetLocalSize
can't be traced in the doxy file. Need to check the correct syntax.

File: docs/usermanual/multinode.rst [hunk 4]
```diff
@@ -147,7 +152,7 @@ When the parallel manager, global matrix or global vector are writing to a file,
...
 ----------------
```
Comment (SwRaw) [inline docs/usermanual/multinode.rst:157]: .. doxygenfunction:: rocalution::ParallelManager::SetGlobalSize
and
.. doxygenfunction:: rocalution::ParallelManager::SetLocalSize
can't be traced in the doxy file, Need to check the correct syntax.

File: docs/usermanual/windows-installation.rst (hunk 2 above)
Comment (ntrost57) [reply docs/usermanual/windows-installation.rst:56]: No, should be https://github.com/ROCmSoftwarePlatform/rocALUTION instead

File: docs/api/solvers.rst [hunk 5]
```diff
@@ -94,24 +102,27 @@ When the initial matrix associated with the solver is on the accelerator, the so
   ls.Solve(rhs, &x);
 
 
-Clear Function and Destructor
+Clear function and destructor
 =============================
+
 The :cpp:func:`rocalution::Solver::Clear` function clears all the data which is in the solver, including the associated preconditioner. Thus, the solver is not anymore associated with this preconditioner.
 
 .. note:: The preconditioner is not deleted (via destructor), only a :cpp:func:`rocalution::Preconditioner::Clear` is called.
```
Comment (SwRaw) [inline docs/api/solvers.rst:110]: "rocalution::Preconditioner::Clear()" cannot be resolved.

File: docs/api/solvers.rst [hunk 6]
```diff
@@ -184,31 +200,35 @@ Algebraic MultiGrid
...
 .. doxygenclass:: rocalution::SAAMG
 .. doxygenfunction:: rocalution::SAAMG::SetCouplingStrength
 .. doxygenfunction:: rocalution::SAAMG::SetInterpRelax
 
-Ruge-Stueben AMG
+Ruge-stueben AMG
 ================
+
 .. doxygenclass:: rocalution::RugeStuebenAMG
 .. doxygenfunction:: rocalution::RugeStuebenAMG::SetCouplingStrength
```
Comment (SwRaw) [inline docs/api/solvers.rst:221]: "rocalution::RugeStuebenAMG::SetCouplingStrength" cannot be resolved.

File: docs/design/functable.rst [hunk 7]
```diff
@@ -1,15 +1,22 @@
...
+
 All matrix operations (except SpMV) require a CSR matrix.
 
-.. note:: If the input matrix is not a CSR matrix, an internal conversion will be performed to CSR format, followed by a back conversion to the previous format after the operation.
-          In this case, a warning message on verbosity level 2 will be printed.
+.. note:: If the input matrix is not a CSR matrix, an internal conversion is performed to CSR format, followed by a back conversion to the previous format after the operation.
+          In this case, a warning message on verbosity level 2 is printed.
 
 ==================================================================================== =============================================================================== ======== =======
 **LocalMatrix function**                                                             **Comment**                                                                     **Host** **HIP**
```
Comment (SwRaw) [inline docs/design/functable.rst:22]: Set​Random​Normal and ExclusiveScan cannot be resolved.

File: docs/usermanual/singlenode.rst [hunk 8]
```diff
@@ -336,11 +345,11 @@ File I/O
 Access
 ======
 
-.. doxygenfunction:: rocalution::LocalVector::operator[](int)
+.. doxygenfunction:: rocalution::LocalVector::&operator[](int)
```
Comment (SwRaw) [inline docs/usermanual/singlenode.rst:349]: rocalution::LocalVector::&operator cannot be resolved.

**Comment by SwRaw** [comment]
There is a lot of repetition of API content. As we call the same doxyfunction at many places. The solvers and preconditioners are individual files while being a part of API as well. I tried to remove the solvers and preconditioners from API library with a reference to the respective file but then realized that there are many cross-references to the solvers/preconditioners in the API library. Removing these would lead to many broken links so I am keeping them. I still feel the entire documentation needs a lot of rework to remove/reduce redundant content. Instead of calling the doxyfunction, call them at one place and add links to all other places where they are needed.

**Comment by SwRaw** [comment]
@lpaoletti Please review.

File: docs/usermanual/windows-installation.rst (hunk 2 above)
Comment (SwRaw) [reply docs/usermanual/windows-installation.rst:56]: Done

**Review by YvanMokwinski — APPROVED** [review]

**Comment by jsandham** [comment]
This is a great start. Approved.

**Review by jsandham — APPROVED** [review]

**Review by ntrost57 — APPROVED** [review]

**Pull request merged**